✨ Kube-bind KubeStellar resources for IMWs, WMWs, and MBWs

waltforme opened the pull request:
## Summary
This PR, along with previous PRs, finishes step 2 of the 'Use kube-bind' section listed in #1208.

Just a partial copy of that list:
1. the KS container image (used by all KS containers) should include the KB needed executables
2. Setup scripts that build the ESPW, WDS, and IS should be changed to:
- Run dex,
- ESPW: Run backend engine + import all KS CRDs + add KB export label + import the KB CRDs
- IS, WDS: Run konnector, bind all KS resources

waltforme commented:
Manual checks for the existence of the KubeStellar CRDs:
```
bash-5.1$ kc ws root:imw1
Current workspace is "root:imw1".
bash-5.1$ kc get crds
NAME                              CREATED AT
apiservicebindings.kube-bind.io   2023-11-01T17:33:01Z
locations.edge.kubestellar.io     2023-11-01T17:33:08Z
synctargets.edge.kubestellar.io   2023-11-01T17:33:06Z
bash-5.1$ kc ws root:wmw1
Current workspace is "root:wmw1".
bash-5.1$ kc get crds
NAME                                 CREATED AT
apiservicebindings.kube-bind.io      2023-11-01T17:33:47Z
customizers.edge.kubestellar.io      2023-11-01T17:33:52Z
edgeplacements.edge.kubestellar.io   2023-11-01T17:33:50Z
```

waltforme commented:
Also checked the CRUD of `edge-placement-c` in `wmw1` to make sure it is synced as expected between `wmw1` and `espw`.

waltforme commented:
/cc @ezrasilvera

ezrasilvera commented on inner-scripts/kubestellar:
```diff
@@ -84,7 +90,8 @@ function ensure_imws() {
+            bind_resource_for_imw "synctargets"
+            bind_resource_for_imw "locations"
             if ! kubectl get apibinding "edge.kubestellar.io" &> /dev/null; then
                 kubectl kcp bind apiexport root:espw:edge.kubestellar.io
```
Do we leave for now both the kcp bind and kube-bind?

waltforme replied:
Yes, that's the current status.

ezrasilvera commented on overlap-scripts/kubectl-kubestellar-ensure-wmw:
```diff
@@ -92,25 +92,32 @@ EOF
 fi
 
 function run_kb_konnector_for_wmw() {
+    if kubectl get ns kube-system &> /dev/null; then
```
This is just a general fix, not related to kube-bind, right ?

waltforme replied:
It **is** related to kube-bind because the konnector needs the kube-system namespace in place otherwise it crashes.

waltforme commented:
Removed the kcp apibinding from apiexport `edge.kubestellar.io` in `root:espw` workspace to IWMs and WMWs.

Also added kube-bound singleplacementslices for WMWs just in case we need it. We can remove it if we found that it is not necessary. Let's see.

ezrasilvera commented on overlap-scripts/kubectl-kubestellar-ensure-mbw:
```diff
@@ -0,0 +1,141 @@
+bind_iff_wanted node.k8s.io
+bind_iff_wanted policy
+bind_iff_wanted scheduling.k8s.io
+bind_iff_wanted storage.k8s.io
```
Who is generating the exports for all the above, and where  ?

ezrasilvera replied:
We discussed that offline

ezrasilvera commented:
@waltforme I just noticed now the usage of export/bind on kube resources (in wmw and now in mb), how are we going to support that with kube-bind ? Who is going to export all those CRDs, ESPW ?  so d we need now to add the CRDs & label setting for all theses resources to there ?
Also in KCP this is done through export/binding of groups, we would need now to individually export/bind all the k8s objects .
